### DISCUSSION TITLE
Debug log for users being denied access

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Capabilities#1297] [Feature Requests]
From a customer who is trying to set up a user account, but that user is being denied access

> 
Can I log the permissions/decision tree around the rejection message in Capabilities? This could help us solve it a lot more quickly than trial and error.

From Kevin:

> 
> It is possible to debug this part of WordPress. You can temporarily replace part of the class-wp-user.php file
> https://core.trac.wordpress.org/browser/trunk/src/wp-includes/class-wp-user.php#L820
> ​​
> Replace​ the section around line 820 with this:
> [https://gist.github.com/stevejburge/c10e08fb71bb3642cdb71fc75cbd9f3e​​​](https://gist.github.com/stevejburge/c10e08fb71bb3642cdb71fc75cbd9f3e)
> 
> This will output to the PHP error log
> 

Could we output something into the UI of Capabilities?

**Comment by olatechpro** [comment]
I'm not sure this is feasible. During our discussion in slack on extractor, everybody agreed we cannot use debug_backtrace() in our development code.

Apart from that, this is better been a snippet for customer because they can simply tailor it to the capability they're targettng than us enabling it for capabilities which will easily kill their server due to debug_backtrace() content been very long and doing the check that's not tailored to the capability been debugged where user reload the page once and then check their error log will be too expensive @stevejburge

**Comment by stevejburge** [comment]
OK, thanks @olatechpro. Let's leave this one open as an idea. Maybe a good solution will appear in the future
